accord-core/src/main/java/accord/local/CommandStores.java

```diff
@@ -715,49 +760,144 @@ protected <O> Cancellable mapReduceConsume(PreLoadContext context, Unseekables<?
         return reduced.begin(mapReduceConsume);
     }
 
+    protected <O> Cancellable mapReduceConsume(PreLoadContext context, StoreSelector selector, MapReduceConsume<? super SafeCommandStore, O> mapReduceConsume)
+    {
+        AsyncChain<O> reduced = mapReduce(context, selector, mapReduceConsume);
+        return reduced.begin(mapReduceConsume);
+    }
+
     public  <O> Cancellable mapReduceConsume(PreLoadContext context, IntStream commandStoreIds, MapReduceConsume<? super SafeCommandStore, O> mapReduceConsume)
     {
         AsyncChain<O> reduced = mapReduce(context, commandStoreIds, mapReduceConsume);
         return reduced.begin(mapReduceConsume);
     }
 
-    private static class StoreSelector extends SimpleBitSet implements IndexedQuadConsumer<Object, Object, Object, Object>, IndexedRangeQuadConsumer<Object, Object, Object, Object>
+    public static class NoStoreSelector implements StoreSelector
+    {
+        private static final NoStoreSelector INSTANCE = new NoStoreSelector();
+
+        @Override
+        public StoreSelector refine(TxnId txnId, @Nullable Timestamp executeAt, Participants<?> participants)
+        {
+            return snapshot -> StoreFinder.find(snapshot, participants)
+                                          .filter(snapshot, participants, txnId.epoch(), (executeAt != null ? executeAt : txnId).epoch())
+                                          .iterator(snapshot);
+        }
+
+        @Override
+        public Iterator<CommandStore> select(Snapshot snapshot)
+        {
+            throw new UnsupportedOperationException();
```

Comment:
been tracing the code and not found 1 place that calls `none` then `select` without `refine`.  Its nested so might have missed something

Reply:
I have made a last minute change to split out these into two interfaces, so we make the transition explicit. We can bike shed naming in post.